src/main/scala/is/hail/io/vcf/ExportVCF.scala

```diff
@@ -1,96 +1,106 @@
 package is.hail.io.vcf
 
-import is.hail.annotations.{Annotation, Querier}
-import is.hail.expr.{Field, TArray, TBoolean, TCall, TFloat64, TFloat32, TGenotype, TInt32, TIterable, TInt64, TSet, TString, TStruct, Type}
+import is.hail
+import is.hail.annotations.MemoryBuffer
+import is.hail.expr._
 import is.hail.utils._
-import is.hail.variant.{Call, GenericDataset, Genotype, Locus, Variant, VariantSampleMatrix}
-import org.apache.spark.sql.Row
+import is.hail.variant.{Genotype, Variant, VariantSampleMatrix}
 
 import scala.io.Source
+import scala.reflect.ClassTag
 
 object ExportVCF {
 
   def infoNumber(t: Type): String = t match {
     case TBoolean(_) => "0"
-    case TArray(elementType, _) => "."
+    case TArray(_, _) => "."
+    case TSet(_, _) => "."
     case _ => "1"
   }
 
-  def strVCF(sb: StringBuilder, elementType: Type, a: Annotation) {
-    if (a == null)
-      sb += '.'
-    else {
-      elementType match {
-        case TFloat32(_) =>
-          val x = a.asInstanceOf[Float]
-          if (x.isNaN)
-            sb += '.'
-          else
-            sb.append(x.formatted("%.5e"))
-        case TFloat64(_) =>
-          val x = a.asInstanceOf[Double]
-          if (x.isNaN)
-            sb += '.'
-          else
-            sb.append(x.formatted("%.5e"))
-        case TInt64(_) =>
-          val x = a.asInstanceOf[Long]
-          if (x > Int.MaxValue || x < Int.MinValue)
-            fatal(s"Cannot convert Long to Int if value is greater than Int.MaxValue (2^31 - 1) or less than Int.MinValue (-2^31). Found $x.")
-          sb.append(elementType.str(x))
-        case _ => sb.append(elementType.str(a))
-      }
+  def strVCF(sb: StringBuilder, elementType: Type, m: MemoryBuffer, offset: Long) {
+    elementType match {
+      case TInt32(_) =>
+        val x = m.loadInt(offset)
+        sb.append(x.toString)
+      case TInt64(_) =>
+        val x = m.loadLong(offset)
+        if (x > Int.MaxValue || x < Int.MinValue)
+          fatal(s"Cannot convert Long to Int if value is greater than Int.MaxValue (2^31 - 1) " +
+            s"or less than Int.MinValue (-2^31). Found $x.")
+        sb.append(x.toInt.toString)
```

Comment:
Same.  Also, no need to `toInt`.  If the long fits, its decimal representation will be fine.